server/controllers/ArticleController.js

```diff
@@ -119,6 +119,97 @@ class ArticleController {
       });
     });
   }
+
+  /**
+   * controller to like or dislike an article
+   * @param {object} req
+   * @param {object} res
+   * @param {object} next
+   * @returns {void}
+   */
+  static async likeOrDislikeArticle(req, res, next) {
+    const { articleId, action } = req.params;
+    const userId = req.userData.id;
+
+    /**
+     * This function likes or dislikes an article
+     * @param {boolean} status - true for like, false for dislike
+     * @returns {void}
+     */
+    const likeDisliker = async (status) => {
+      let articleLike;
+      let created;
+      try {
+        // find the article like or create it
+        [articleLike, created] = await ArticleLike.findOrCreate({
+          where: { [Op.and]: [{ userId }, { articleId }] },
+          defaults: { status, articleId, userId, }
+        });
+      } catch (err) {
+        return next(err);
+      }
+
+      // suffix for message
+      const suffix = status ? 'like' : 'dislike';
+      // get the article like id
+      const articleLikeId = articleLike.dataValues.id;
+
+      if (created) {
+        // if like/dislike was added
+        return res.status(201).json({
+          status: 'success',
+          message: `article successfully ${suffix}d`,
+        });
+      }
+
+      // if like/dislike was not added
+      if (status !== articleLike.dataValues.status) {
+        // opposite action was triggered, update the like
+        const [rowCount] = await ArticleLike.update({
+          status: !articleLike.dataValues.status,
+        }, {
+          where: {
+            id: articleLikeId,
+          }
+        });
+        if (rowCount > 0) {
+          return res.status(200).json({
+            status: 'success',
+            message: `you changed your mind, article successfully ${suffix}d`,
+          });
+        }
+      }
+      // same action was triggered, undo the like or dislike
+      const undoRows = await ArticleLike.destroy({
+        where: {
+          id: articleLikeId,
+        }
+      });
+      if (undoRows > 0) {
+        return res.status(200).json({
+          status: 'success',
+          message: `article ${suffix}, undo successful`,
+        });
+      }
+    };
+
+    // error for unknown action in switch statement
+    // eslint-disable-next-line max-len
```

Comment:
I observe you disabled the max-len rule. The max len rule in the eslintrc file is to ensure our codebase is readable. Can you justify why you disable it here?

Reply:
@sulenchy I agree that the reason for linting is to make our code more readable. However, I have disabled the `max-len` rule in this one line for the following reason:
- Following the airbnb style guide, it states not to write long strings over multiple lines to make them easier to work with. see here —> https://github.com/airbnb/javascript#strings.